### PR TITLE
Add and demo ConnectionSettings and ContinueButton components

### PR DESCRIPTION
**Pull request by jarolrod**
This introduces the `ConnectionSettings` component, which represents the [detailed connections settings page](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App-Main?node-id=2777%3A67296) as designed in the [main design file](https://www.figma.com/file/GaCoOSNHB2yMB9ThiDtred/Bitcoin-Core-App-Main?node-id=1035%3A1883). It also introduces the controls the component depends on, `Setting` and `OptionSwitch`.  Missing is the Text button that is used for settings which require an input value, because the functionality of that component needs to be discussed.

The `Setting` control represents an individual setting of the `ConnectionSettings` components, this component uses the `Header` control for its text. `OptionSwitch` represents the toggle button shown in the design.

To accomplish a proper demo of this component, this places our forms (ConnectionOptions and ConnectionSettings) into a StackLayout. The `ContinueButton` is introduced to advance/rewind the stack. A `PageIndicator` is added to highlight the functionality of the demo.

Additionally, to fix the annoying behavior of the stub currently, this makes the dimensions of the stub page dependent on it's contents width. 😄 

**Image of added components:**
Screen Shot 2022-01-03 at 9 28 59 PM

[Windows](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/101)
[macOS](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/101)
[Android](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/101)

### CODE REVIEW

**Comment by jarolrod** [comment]
updated from 8f801eb -> 533993f 

Changes:
- rebased over master
- updated include versions
- Settings.qml
  - Setting control base type changed from Item to control
  - removed instances of `height: sourceComponent.height`
  - removed setting `implicitHeight`
- dropped change on the stubs size as it's not relevant to the goal of this PR

**Comment by jarolrod** [comment]
updated from 5d10e90 -> e12442b

Changes: 
- in settings.qml bind visible to active

File: src/qml/pages/stub.qml [hunk 1]
```diff
@@ -52,9 +52,21 @@ ApplicationWindow {
                 onClicked: indicator.progress = mouseX / width
             }
         }
-        ConnectionOptions {
-            Layout.preferredWidth: 400
-            focus: true
+        StackLayout {
```
Comment (promag) [inline src/qml/pages/stub.qml:55]: FYI stack doesn't support animation.

Comment (hebasto) [reply src/qml/pages/stub.qml:55]: Is it a blocker for this PR?

Comment (promag) [reply src/qml/pages/stub.qml:55]: No.

Comment (jarolrod) [reply src/qml/pages/stub.qml:55]: yeah it's just used here for simplicity of the demo. #106 uses an uninteractive SwipeView which has the animations we want.

**Review by hebasto — APPROVED** [review]
ACK e12442b5777ebef8e0e07c5314b1671367c8d5fd, tested on Linux Mint 20.3 (with depends).

~As this PR has no conflicts with #110, going to merge it.~

**Comment by hebasto** [comment]
Could the last commit (a demo) be dropped from this PR, to avoid conflicts with #110?

**Comment by jarolrod** [comment]
updated from e12442b -> c026a3e

Changes: drop last commit to avoid conflict with #110. The demo of this PR has been moved to: [connection-settings-demo](https://github.com/jarolrod/gui-qml/tree/connection-settings-demo)

**Review by hebasto — APPROVED** [review]
re-ACK c026a3e85c8e21519c85b840a4ee8e8d444eb88d

File: src/qml/components/ConnectionSettings.qml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import "../controls"
+
+ColumnLayout {
+    spacing: 20
+    Setting {
+        Layout.fillWidth: true
+        header: "Use cellular data"
```
[the remark this answers is not in the document]
Comment (shaavan) [reply src/qml/components/ConnectionSettings.qml:14]: Addressed in #111

File: src/qml/controls/Setting.qml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+Control {
+    id: root
+    property bool last: false
```
[the remark this answers is not in the document]
Comment (shaavan) [reply src/qml/controls/Setting.qml:11]: Addressed in #111

File: src/qml/controls/OptionSwitch.qml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+
+Switch {
+    id: root
+    indicator: Rectangle {
+        implicitWidth: 45
+        implicitHeight: 28
+        x: root.leftPadding
+        y: parent.height / 2 - height / 2
```
[the remark this answers is not in the document]
Comment (shaavan) [reply src/qml/controls/OptionSwitch.qml:14]: Addressed in #111

File: src/qml/controls/Setting.qml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    property string header
+    property string description
+    contentItem: GridLayout {
```
[the remark this answers is not in the document]
Comment (shaavan) [reply src/qml/controls/Setting.qml:14]: What would be a more appropriate Layout for it?

[the remark this answers is not in the document]
Comment (jarolrod) [reply src/qml/controls/Setting.qml:14]: A ColumnLayout with a nested RowLayout

**Pull request merged**